File: Data Structures & Algorithms/depth-of-binary-tree/submission-0.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

public class Solution {
    public int MaxDepth(TreeNode root) {
        
        return FindDepth(root, 0);
        
    }
    int FindDepth(TreeNode node, int depth){
        if(node == null){
            return depth;
        }
        depth++;
        int Ldepth = FindDepth(node.left, depth);
        int Rdepth = FindDepth(node.right, depth);
        return Math.Max(Ldepth, Rdepth);
    }
}

File: Data Structures & Algorithms/binary-tree-from-preorder-and-inorder-traversal/submission-0.js
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     constructor(val = 0, left = null, right = null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    /**
     * @param {number[]} preorder
     * @param {number[]} inorder
     * @return {TreeNode}
     */
    buildTree(preorder, inorder) {
        if(preorder.length == 0 || inorder.length == 0){
            return null;
        }
        const root = new TreeNode(preorder[0]);
        let targetIndex = inorder.indexOf(root.val);
        root.left = this.buildTree(preorder.slice(1, targetIndex + 1), inorder.slice(0, targetIndex));
        root.right = this.buildTree(preorder.slice(targetIndex + 1), inorder.slice(targetIndex + 1));
        
        return root;


    }
}
